components/compliance-service/api/tests/010_controls_list_spec.rb

```diff
@@ -356,4 +407,217 @@ def reporting
+                                        "waived" => {
+                                            "total" => 1
+                                        }
+                                    }
+                            }
+                        ]
+                },
+                {
+                    "id" => "pro1-con5",
+                    "title" => "Profile 1 - Control 5",
+                    "profile" => {
+                        "title" => "My Profile 1 title",
+                        "id" => "447542ecfb8a8800ed0146039da3af8fed047f575f6037cfba75f3b664a97ea4",
+                        "version" => "1.0.1"
+                    },
+                    "impact" => 0.89999998,
+                    "endTime" => "2018-04-01T10:18:41Z",
+                    "controlSummary" => {
+                        "total" => 1,
+                        "passed" => {},
+                        "skipped" => {},
+                        "failed" => {
+                            "total" => 1,
+                            "critical" => 1
+                        },
+                        "waived" => {}
+                    }
+                },
+                {
+                    "id" => "pro2-con1",
+                    "title" => "Profile 2 - Control 1",
+                    "profile" => {
+                        "title" => "My Profile 2 title",
+                        "id" => "447542ecfb8a8800ed0146039da3af8fed047f575f6037cfba75f3b664a97ea5",
+                        "version" => "1.0.5"
+                    },
+                    "impact" => 0.80000001,
+                    "endTime" => "2018-04-01T10:18:41Z",
+                    "controlSummary" => {
+                        "total" => 2,
+                        "passed" => {},
+                        "skipped" => {},
+                        "failed" => {},
+                        "waived" => {
+                            "total" => 2
+                        }
+                    },
+                    "waivers" =>
+                        [
+                            {
+                                "waivedStr" => "yes_run",
+                                "justification" => "Sound reasoning",
+                                "waiverSummary" =>
+                                    {
+                                        "passed" => {},
+                                        "skipped" => {},
+                                        "failed" => {
+                                            "total" => 2
+                                        },
+                                        "waived" => {
+                                            "total" => 2
+                                        }
+                                    }
+                            }
+                        ]
+                }
+            ]
+    }
+    control_item_array = actual_data['control_items']
+    assert_equal(6, control_item_array.size)
```

Comment:
Testing the size of the `control_items` like this is going to give less information on a test failure than the next test will. The next test's output will be much more verbose, but I feel like that's a good thing in an integration test.

Reply:
I agree @phiggins. The second test implicitly tests the counts so no need for both.. took out count test.  Thanks